### PR TITLE
remove fake override declarations from declaration container descriptor implementations

### PR DESCRIPTION
**Pull request by neetopia**

### CODE REVIEW

**Comment by neetopia** [comment]
cc @yigit

File: compiler-plugin/testData/api/declared.kt [hunk 1]
```diff
@@ -0,0 +1,26 @@
+// TEST PROCESSOR: DeclaredProcessor
+// EXPECTED:
+// Base class delcared functions:
```
Comment (yigit) [inline compiler-plugin/testData/api/declared.kt:3]: nit: declared

File: compiler-plugin/testData/api/declared.kt [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+}
+// MODULE: main(module1)
+// FILE: sub.kt
+class Sub: Base() {
+    fun subFun() {}
+}
+
+// FILE: JavaSource.java
+public class JavaSource {
+    public int javaSourceFun {
```
Comment (yigit) [inline compiler-plugin/testData/api/declared.kt:23]: nit: missing ()

File: compiler-plugin/testData/api/declared.kt [hunk 3]
```diff
@@ -0,0 +1,26 @@
+// TEST PROCESSOR: DeclaredProcessor
+// EXPECTED:
+// Base class delcared functions:
+// subFun
+// Sub class delcared functions:
+// baseFun
+// JavaSource class delcared functions:
+// javaSourceFun
+// END
+// MODULE: module1
```
Comment (yigit) [inline compiler-plugin/testData/api/declared.kt:10]: yay, happy to see the module-test infra being useful :)

**Review by ting-yuan — APPROVED** [review]
LGTM. Please merge after addressing Yigit's comments.

**Pull request merged**